app/models/metric/ci_mixin/capture.rb

```diff
@@ -49,7 +49,7 @@ def perf_capture_queue(interval_name, options = {})
     cb = nil
     if interval_name == 'historical'
       start_time = Metric::Capture.historical_start_time if start_time.nil?
-      end_time = Time.now.utc if end_time.nil?
+      end_time = Time.now.utc.end_of_day # Ensure no more than one historical collection is queue up in the same day
```

Comment:
Also, should this be an `||=`.  I and end_time is passed in, why wouldn't we honor it?